predicate/predicate_slots.go

```diff
@@ -9,15 +9,15 @@ import (
 	"github.com/ava-labs/libevm/core/types"
 )
 
-type predicaters interface {
+type PredicaterExistChecker interface {
```

Comment:
doesn't seem this type needs to be exported

Reply:
Correct; I don't know why I kept exporting argument types of exported functions, I must had read somewhere a reason for it but I cannot find it again 🤷  Unexporting ⛵  !

Reply:
Actually, re-exporting!

> As a general rule I'd say yes because it's kind to the users of your package.

I did read this as well somewhere, and, although it goes against encapsulation, it does help the consumer of the exported function. For example godoc will be generated on that type as well, whereas it would not if unexported.

Reply:
It also allows a provider of the type to locally assert that they implement it, in an idiomatic way.

```go
var _ predicate.PredicaterExistChecker = (*myImpl)(nil) // idiomatic

var _ = predicate.PreparePredicateStorageSlots((*myImpl)(nil), ...) // wtf? might also panic
```